ucfopen/Obojobo
# Adds assessment stats for modules and new /stats page in the repository

**⚠️ Note: This branch has migrations and new packages! ⚠️**

* Added canViewSystemStats permission. By default this stat can't be added via an LTI launch, it must be applied manually
* Added user_perms table, which unilaterally applies a permission (like canViewSystemStats) to a user
* Added a new Assessment Stats dialog to modules in the repository. This allows you to see a score breakdown of students' final assessment scores and their attempt scores.
* Added /api/assessments/:draftId/details, a new API endpoint to power the new assessment stats page
* Added a new /stats page to the repository, which allows users with the canViewSytemStats permission to be able to view assessment details for any draft in the system
* Added /api/drafts-all, which returns draft summary information for all drafts in the system

Additional things:
* Adds the year to the version listing on the version history dialog
* Changed the name of AssessmentModel's fetchAttempts to fetchAttemptsForUserDraftAndResourceLinkId
* Changes User so that instead of methods like user.canPreviewDrafts which are created dynamically, now user has a perms array, and you would use user.hasPermission('canPreviewDrafts')
* Removes the 'Dashboard' link from the repository if the user isn't logged in

To test:
* Click on the More... button for a module in the repository, then click on the new Assessment Stats button
* Add a record in user_perms for the Test Instructor 1 user, with the permission `{ canViewSystemStats }` - This should add a new Stats option in the repository header nav bar
